### PR TITLE
docs: advanced view features (structured projections and joins)

### PR DESCRIPTION
**Pull request by pvlugter**
Refs https://github.com/lightbend/kalix/issues/7323

Document some of the new advanced view features: structured projections and joins with multiple tables. Extend the customer registry example used in views to a simple store example with products and orders.

Adds an advanced view queries section, in a separate partial so it can be shared with Spring docs. Spring SDK will need some changes to support multiple tables, so will follow up with that separately.

These additional view features are not available by default. There's a note about this, and how to enable for local proxy and integration tests.

~Proxy version bump included here for now. Marking as draft until proxy release is finalised.~

### CODE REVIEW

**Comment by pvlugter** [comment]
Should be ready now, with proxy 1.0.28.

**Comment by pvlugter** [comment]
Realising that there are deeper changes needed for Views in the SDKs — given all the assumptions around a View having a single state type. So views with joins that don't transform work fine, but views that transform updates are not expressible in code right now — or more, only update transforms for the primary table would work as that's the single state type.

Switching the docs back to draft again, until some SDK changes are figured out, and for code generation and Spring SDK.

**Comment by octonato** [comment]
> Realising that there are deeper changes needed for Views in the SDKs — given all the assumptions around a View having a single state type. So views with joins that don't transform work fine, but views that transform updates are not expressible in code right now — or more, only update transforms for the primary table would work as that's the single state type.
> 
> Switching the docs back to draft again, until some SDK changes are figured out, and for code generation and Spring SDK.

I just realised that this will be a problem for the Spring SDK. At some point we supported event handlers that would receive the current view state and the incoming event or value change. 

We reverted that option because we found it confusing and error prone. Also, we want it to be more aligned with the other components where we have a `currentState` method returning it. So, for views, we introduced a `viewState()` method. Well, that won't work anymore.

**Review by johanandren — COMMENTED** [review]
Didn't review the sample in detail but rendered the docs locally and read through, the doc section looks good 👍

**Comment by pvlugter** [comment]
Continuing ideas for supporting multi-table views with update transforms in https://github.com/lightbend/kalix/issues/7323#issuecomment-1321281453

**Comment by pvlugter** [comment]
Updated samples to have customers and products be event sourced entities, with transform updates in the views, and orders to still be a value entity. Builds on the multi-table view support in #1299, rebased on that branch temporarily.

**Comment by johanandren** [comment]
I mixed up this PR with #1299, so consider my review a review of both.

File: sdk/scala-sdk/src/main/scala/kalix/scalasdk/view/View.scala [hunk 1]
```diff
@@ -99,6 +99,6 @@ abstract class View[S] {
    *   an empty state object or `null` to hand to the process method when an event for a previously unknown subject id
    *   is seen.
    */
-  def emptyState: S
+  def emptyState: S = null.asInstanceOf[S]
```
[the remark this answers is not in the document]
Comment (pvlugter) [reply sdk/scala-sdk/src/main/scala/kalix/scalasdk/view/View.scala:102]: Makes sense. I'll restore in #1299. And update the comment, which mentions null.

[the remark this answers is not in the document]
Comment (pvlugter) [reply sdk/scala-sdk/src/main/scala/kalix/scalasdk/view/View.scala:102]: Restored in https://github.com/lightbend/kalix-jvm-sdk/pull/1299/commits/b4ccc3945b67bf895aad8279560282f9575e9660

**Comment by pvlugter** [comment]
Integration tests for the views seem prone to timing out. Already a 20 second timeout on waiting for the view to update, but will increase a bit more.

**Comment by pvlugter** [comment]
Can still fail with 60 second timeout. Only the Java version does this. Seems strange — is the different awaitility?

**Comment by pvlugter** [comment]
Tracked down the integration test issue to a recently introduced dev proxy bug for in-memory journal. Fixed in https://github.com/lightbend/kalix-proxy/pull/1746.

**Comment by pvlugter** [comment]
Bumped to proxy 1.0.29 for integration tests. No other changes since review.

**Pull request merged**